src/SetProtocol.ts

```diff
@@ -301,9 +301,48 @@ class SetProtocol {
    * @param componentAddresses    Component ERC20 addresses
    * @return                      The minimum value of the natural unit allowed by component decimals
    */
-  public async calculateMinimumNaturalUnit(components: Address[]): Promise<BigNumber> {
+  public async calculateMinimumNaturalUnitAsync(components: Address[]): Promise<BigNumber> {
     return await this.factory.calculateMinimumNaturalUnit(components);
   }
+
+  /**
+   * Calculates the component units required to achieve a target Set price given a list of
+   * components, desired component proportions (in decimal format), and component prices.
+   *
+   * @param componentPrices         A list of fiat-denominated component prices
+   * @param componentAddresses      Component ERC20 addresses
+   * @param componentProportions    Decimal-formatted allocations. Must add up to 1
+   * @param targetSetPrice          Desired price of a single unit of the Set
```

Comment:
Specify USD?

Reply:
It can be in any denomination the developer chooses - though I will say `fiat-denominated price`